src/player.cpp

```diff
@@ -1159,7 +1160,53 @@ void player::update_bodytemp()
         }
         // Loss of blood results in loss of body heat, 1% bodyheat lost per 2% hp lost
         temp_conv[bp] -= blood_loss( bp ) * temp_conv[bp] / 200;
-
+        bool waterproof_shoes = false;
+        for( const item &worn_item : worn ) {
+            if (worn_item.covers(bp) && worn_item.has_flag("WATERPROOF") ){
+                waterproof_shoes = true;
+            }
+        }
+        bool wearing_shoes = is_wearing_shoes( side::LEFT ) || is_wearing_shoes( side::RIGHT );
+        const bool mutfeet = has_trait( trait_LEG_TENTACLES ) || has_trait( trait_PADDED_FEET ) ||
+                             has_trait( trait_HOOVES ) || has_trait( trait_TOUGH_FEET ) || has_trait( trait_ROOTS2 );
+        switch( bp ) {
+            case bp_eyes:
+            case bp_head:
+            case bp_torso:
+            case bp_mouth:
+            case bp_leg_l:
+            case bp_leg_r:
+            case bp_arm_l:
+            case bp_arm_r:
+            case bp_hand_l:
+            case bp_hand_r:
+                break;
+            case bp_foot_l:
+                if( snowonground ){
+                    if( waterproof_shoes ){
+                        break;
+                    } else if( wearing_shoes || mutfeet ){
+                        temp_conv[bp] -= 300;
+                    } else {
+                        temp_conv[bp] -= 450;
+                    }
+                }
+                break;
+            case bp_foot_r:
+                if( snowonground ){
+                    if( waterproof_shoes ){
+                        break;
+                    } else if( wearing_shoes || mutfeet ){
+                        temp_conv[bp] -= 300;
+                    } else {
+                        temp_conv[bp] -= 450;
+                    }
+                }
+                break;
```

Comment:
Duplicated cases:
```c++
case bp_foot_l:
case bp_foot_r:
    ...
```
You probably don't want a gigantic `switch` statement here since you're only interested in feet. A couple of conditions in an `if` would suffice.